packages/ui/src/components/ConfirmModal/index.tsx

```diff
@@ -0,0 +1,56 @@
+import { I18nKey } from '@logto/phrases';
+import classNames from 'classnames';
+import React, { ReactNode } from 'react';
+import { useTranslation } from 'react-i18next';
+import ReactModal from 'react-modal';
+
+import Button from '@/components/Button';
+import * as modalStyles from '@/scss/modal.module.scss';
+
+import * as styles from './index.module.scss';
+
+type Props = {
+  classname?: string;
+  isOpen?: boolean;
+  children: ReactNode;
+  cancelText?: I18nKey;
+  confirmText?: I18nKey;
+  onConfirm?: () => void;
+  onClose: () => void;
+};
+
+const ConfirmModal = ({
+  classname,
+  isOpen = false,
+  children,
+  cancelText,
+  confirmText,
+  onConfirm,
+  onClose,
+}: Props) => {
+  const { t } = useTranslation();
+
+  return (
+    <ReactModal
+      role="dialog"
+      isOpen={isOpen}
+      className={classNames(modalStyles.modal, classname)}
+      overlayClassName={modalStyles.overlay}
+      parentSelector={() => document.querySelector('main') ?? document.body}
+    >
+      <div className={styles.container}>
+        <div className={styles.content}>{children}</div>
+        <div className={styles.footer}>
+          <Button type="secondary" size="small" onClick={onClose}>
+            {t(cancelText ?? 'general.cancel')}
```

Comment:
Will `t(cancelText)` work?

Reply:
hmm good call should give a default value

Reply:
@demonzoo updated PTAL